version.py

```diff
@@ -80,7 +78,7 @@ def git_describe_version():
       after the most recent tag(v0.7.0),
       the git short hash tag of the current commit is 0d07a329e.
     """
-    cmd = ["git", "describe", "--tags", "--match", "v[0-9]*.[0-9]*.[0-9]*"]
+    cmd = ["git", "describe", "--tags", "--match", "v[0-9]*.[0-9]*.[dev0-9]*"]
```

Comment:
The glob pattern here matches also, for instance, `v0.7.ev0`, because `dev` is not treated as a string, but as a char set. If both exact tags (like v0.7.0) and dev tags must be matched I think there is not way to do that besides adding a `--match "v[0-9]*.[0-9]*.dev[0-9]*"` after the exiting `--match`, since multiple `--match` are accumulated. I'm a tad surprised git is not using something more powerful (regex) for these patterns... : /

Reply:
Correct, excellent suggestion!